test/test-spawn.c

```diff
@@ -1394,6 +1394,67 @@ TEST_IMPL(spawn_no_path) {
   MAKE_VALGRIND_HAPPY(uv_default_loop());
   return 0;
 }
+
+
+TEST_IMPL(spawn_no_ext) {
+  char new_exepath[1024];
+
+  init_process_options("spawn_helper1", exit_cb);
+  strncpy(new_exepath, exepath, exepath_size - (sizeof(".exe") / sizeof(char) - 1));
+  strncpy(new_exepath + exepath_size - (sizeof(".exe") / sizeof(char) - 1), "_no_ext", sizeof("_no_ext") / sizeof(char));
```

Comment:
strncpy is very confusing and hard to use right (unlike other APIs with this name, it overwrites all `n` bytes, rather than adding safety). A memcpy would be clearer here